src/components/data-exploration/heatmap/HeatmapPlot.jsx

```diff
@@ -33,22 +48,31 @@ const HeatmapPlot = (props) => {
   const loadingGenes = useSelector((state) => state.genes.expression.loading);
   const selectedGenes = useSelector((state) => state.genes.expression.views[COMPONENT_TYPE]?.data);
 
-  const [vegaData, setVegaData] = useState(null);
-  const [vegaSpec, setVegaSpec] = useState(spec);
+  const [viewState, setViewState] = useState({ zoom: 0, target: [0, 0] });
+  const [heatmapData, setHeatmapData] = useState(null);
+  const [highlightedTrackData, setHighlightedTrackData] = useState(null);
+
   const [isHeatmapGenesLoading, setIsHeatmapGenesLoading] = useState(false);
-  const currentHeatmapSettings = useRef();
 
+  const [geneHighlight, setGeneHighlight] = useState(null);
+  const [cellHighlight, setCellHighlight] = useState(null);
+
+  const [maxCells, setMaxCells] = useState(1000);
```

Comment:
what do we use this maxCells for?

Reply:
At the time it was probably related to downsampling I woudl assume, but now it doesn't seem to be doing anything

Reply:
At the time it was probably related to downsampling I woudl assume, but now it doesn't seem to be doing anything

Reply:
how about we remove it then?

Reply:
If you remove it, you will also need to remove this useEffect:

```
  useEffect(() => {
    setMaxCells(Math.floor(width * 0.8));
  }, [width]);
```